cluster/engine.go

```diff
@@ -184,6 +185,14 @@ func (e *Engine) StartMonitorEvents() {
 func (e *Engine) ConnectWithClient(client dockerclient.Client) error {
 	e.client = client
 
+	// Use HTTP Client used by dockerclient to create engine-api client
```

Comment:
I think it better to move this new code to `Connect()` method where client is initialized.